ProcessMaker/processmaker
# [FOUR-15330] Refactor column and advanced filter state when editing inbox rules

## Issue & Reproduction Steps
See https://processmaker.atlassian.net/browse/FOUR-15330

When creating or editing an inbox rule, column changes are not being saved and do not re-load the task list.

## Solution
- Move the state of the columns up to the Edit.vue component, as well as a number of other refactors

## How to Test
- See steps in jira issue
- Also, add a new custom column for a data field. Save and re-open the inbox rule to ensure its still showing.

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-15330

## Code Review Checklist

ci:next
.
